cilium-cli/connectivity/builder/pod_to_pod_encryption.go

```diff
@@ -25,17 +26,17 @@ func (t podToPodEncryption) build(ct *check.ConnectivityTest, _ map[string]strin
 			tests.PodToPodEncryption(features.RequireEnabled(features.EncryptionPod)),
 		)
 
-	newTest("pod-to-pod-with-l7-policy-encryption", ct).
-		WithCondition(func() bool { return !ct.Params().SingleNode }).
-		WithFeatureRequirements(
-			features.RequireEnabled(features.L7Proxy),
-			// Once https://github.com/cilium/cilium/issues/33168 is fixed, we
-			// can enable for IPsec too.
-			features.RequireMode(features.EncryptionPod, "wireguard"),
-		).
-		WithCiliumPolicy(clientsEgressL7HTTPFromAnyPolicyYAML).
-		WithCiliumPolicy(echoIngressL7HTTPFromAnywherePolicyYAML).
-		WithScenarios(
-			tests.PodToPodEncryption(features.RequireEnabled(features.EncryptionPod)),
-		)
+	for _, encryption := range []string{"wireguard", "ipsec"} {
+		newTest(fmt.Sprintf("pod-to-pod-with-l7-policy-encryption-%s", encryption), ct).
+			WithCondition(func() bool { return !ct.Params().SingleNode }).
+			WithFeatureRequirements(
+				features.RequireEnabled(features.L7Proxy),
+				features.RequireMode(features.EncryptionPod, encryption),
+			).
+			WithCiliumPolicy(clientsEgressL7HTTPFromAnyPolicyYAML).
+			WithCiliumPolicy(echoIngressL7HTTPFromAnywherePolicyYAML).
+			WithScenarios(
+				tests.PodToPodEncryption(features.RequireEnabled(features.EncryptionPod)),
+			)
+	}
```

Comment:
drive-by - we'll need to guard the IPsec part here to only run against v1.17 releases, until we've backported the fix into stable branches. Otherwise the test will for example run against an unfixed `v1.14` branch, and fail.

It probably also makes sense to split this change into two patches (Cilium agent vs CLI), so that it's easier to drop the CLI change when backporting.

Reply:
You're right, thanks for the catch!